File: mvc-architecture/mvc-arch/src/main/java/org/example/mvc/repository/ConnectionManager.java
package org.example.mvc.repository;

import com.zaxxer.hikari.HikariDataSource;

import javax.sql.DataSource;
import java.sql.Connection;
import java.sql.SQLException;

public class ConnectionManager {
    private static final String dbName = "com.mysql.cj.jdbc.Driver";
    private static final String url = "jdbc:mysql://localhost:3306/mvc";
    private static final String username = "root";
    private static final String password = "1111";
    private static final Integer maxPoolSize = 40;
    private static DataSource ds;

    private static DataSource getDataSource() {
        HikariDataSource hikariDataSource = new HikariDataSource();

        hikariDataSource.setDriverClassName(dbName);
        hikariDataSource.setJdbcUrl(url);
        hikariDataSource.setUsername(username);
        hikariDataSource.setPassword(password);
        hikariDataSource.setMaximumPoolSize(maxPoolSize);

        return hikariDataSource;
    }

    public static Connection getConnection() throws SQLException {
        if (ds == null) ds = getDataSource();
        return ds.getConnection();
    }
}

File: FastCampus/jdbc/src/main/kotlin/org/example/ConnectionManager.kt
package org.example

import com.zaxxer.hikari.HikariDataSource
import java.sql.Connection
import javax.sql.DataSource


class ConnectionManager {
    companion object {
        private const val dbName = "org.postgresql.Driver"
        private const val url = "jdbc:postgresql://localhost:5432/jdbc_test"
        private const val username = "postgres"
        private const val password = "1111"
        private const val maxPoolSize = 40
        private var ds: DataSource? = null

        internal fun getDataSource(): DataSource {
            val hikariDataSource = HikariDataSource()
            hikariDataSource.driverClassName = dbName
            hikariDataSource.jdbcUrl = url
            hikariDataSource.username = username
            hikariDataSource.password = password
            hikariDataSource.maximumPoolSize = maxPoolSize
            hikariDataSource.minimumIdle = maxPoolSize

            return hikariDataSource
        }

        internal fun getConnection(): Connection {
            if (ds == null) ds = getDataSource()
            return ds!!.connection
        }
    }
}